Source/JavaScriptCore/b3/B3LowerToAir32_64.cpp

```diff
@@ -1661,7 +1659,10 @@ class LowerToAir {
                 break;
             case ValueRep::SomeRegister:
             case ValueRep::SomeLateRegister:
-                arg = tmp(value.value());
+                if (is32Bit() && value.value()->type() == Int64)
+                    arg = someArg(value.value());
+                else
+                    arg = tmp(value.value());
```

Comment:
Minor nit: I think someArg can handle both cases here. Also is32Bit() should always be true in this file.